internal/scanner/scanner.go

```diff
@@ -863,12 +863,11 @@ func (s *Scanner) Scan() ast.Kind {
 				}
 				s.pos--
 			}
-			if s.scanIdentifier(1) {
-				s.token = ast.KindPrivateIdentifier
-			} else {
+			if !s.scanIdentifier(1) {
 				s.errorAt(diagnostics.Invalid_character, s.pos-1, 1)
-				s.token = ast.KindUnknown
+				s.tokenValue = "#"
 			}
+			s.token = ast.KindPrivateIdentifier
```

Comment:
Seems fine, but why did we need this? Or, why is Strada able to get away without this?

Reply:
The new code does what Strada does. The code before this PR seems like a porting bug.